.github/workflows/ci.yml

```diff
@@ -0,0 +1,32 @@
+name: CI Workflow
+
+on:
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+
+    steps:
+        # Checkout the repository with submodules
+        - name: Checkout repository
+          uses: actions/checkout@v3
+          with:
+            submodules: recursive # Ensures submodules are fetched
+
+        # Run the submodule validation script
+
+        - name: Validate submodule state
+          run: |
+            cd ugbio_utils
+            git fetch origin main
+            LATEST_MASTER_COMMIT=$(git rev-parse origin/main)
+            CURRENT_SUBMODULE_COMMIT=$(git rev-parse HEAD)
+            if [ "$LATEST_MASTER_COMMIT" != "$CURRENT_SUBMODULE_COMMIT" ]; then
+                echo "Submodule ugbio_utils is not up-to-date with the master branch."
+                echo "Expected: $LATEST_MASTER_COMMIT, Found: $CURRENT_SUBMODULE_COMMIT"
```

Comment:
Do we want ugbio to be on main latest commit by default? 
personally , I would like this behavior for development. 
maybe as an open-source repo, ugbio submodule needs to be on latest release tag ?

Reply:
either way, it is nice to explicitly know on what branch ugbio submodule is set on.

Reply:
Agree that this is a valid question. For now let's use the convention that the "master" of VariantCalling points to the "main" of ugbio_utils. If it is becoming compllicated, we can change. However, I believe that it is always a good practice when we develop a feature in VariantCalling that also affects ugbio_utils to first merge the ugbio_utils, then update the VariantCalling's PR to ugbio_utils main. This CI feature just "reminds" one to do this.